compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5624,10 +5624,358 @@ OMR::ARM64::TreeEvaluator::arraysetEvaluator(TR::Node *node, TR::CodeGenerator *
+         }
+      }
+   else
+      {
+      generateCompareInstruction(cg, node, data1Reg, data2Reg, true);
+      generateCondTrg1Src2Instruction(cg, TR::InstOpCode::cselx, node, data1Reg, data1Reg, data3Reg, TR::CC_NE);
+      generateCondTrg1Src2Instruction(cg, TR::InstOpCode::cselx, node, data2Reg, data2Reg, data4Reg, TR::CC_NE);
+      generateTrg1Src1Instruction(cg, TR::InstOpCode::revx, node, data1Reg, data1Reg);
+      generateTrg1Src1Instruction(cg, TR::InstOpCode::revx, node, data2Reg, data2Reg);
+      }
+   srm->reclaimScratchRegister(data3Reg);
+   srm->reclaimScratchRegister(data4Reg);
+
+   if (!isLengthGreaterThan15)
+      {
+      auto branchToDone0LabelInstr = generateLabelInstruction(cg, TR::InstOpCode::b, node, done0Label);
+
+      auto lessThan16LabelInstr = generateLabelInstruction(cg, TR::InstOpCode::label, node, lessThan16Label);
+      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::subsimmw, node, lengthReg, lengthReg, 1);
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrbpost, node, data1Reg, TR::MemoryReference::createWithDisplacement(cg, src1Reg, 1));
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrbpost, node, data2Reg, TR::MemoryReference::createWithDisplacement(cg, src2Reg, 1));
+      generateConditionalCompareInstruction(cg, node, data1Reg, data2Reg, 0, TR::CC_HI);
+      auto branchBacktoLessThan16LabelInstr = generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, lessThan16Label, TR::CC_EQ);
+      if (debugObj)
+         {
+         debugObj->addInstructionComment(branchToDone0LabelInstr, "Jumps to done0Label.");
+         debugObj->addInstructionComment(lessThan16LabelInstr, "lessThan16Label");
+         debugObj->addInstructionComment(branchBacktoLessThan16LabelInstr, "Jumps to lessThan16Label (byteloop) if the remaining length > 0 and no mismatch is found");
+         }
+      if (isArrayCmpLen)
+         {
+         generateCompareInstruction(cg, node, data1Reg, data2Reg);
+
+         TR::Register *offReg = srm->findOrCreateScratchRegister();
+         generateCSetInstruction(cg, node, offReg, TR::CC_NE);
+         auto adjustSrc1AddrInstr = generateTrg1Src2Instruction(cg, TR::InstOpCode::subx, node, src1Reg, src1Reg, offReg);
+         if (debugObj)
+            {
+            debugObj->addInstructionComment(adjustSrc1AddrInstr, "Subtracts 1 from src1 if the mismatch is found in the last byte of data.");
+            }
+         }
+      }
+
+   if (isArrayCmpLen)
+      {
+      auto done0LabelInstr = generateLabelInstruction(cg, TR::InstOpCode::label, node, done0Label);
+      generateTrg1Src2Instruction(cg, TR::InstOpCode::subx, node, resultReg, src1Reg, savedSrc1Reg);
+      if (debugObj)
+         {
+         debugObj->addInstructionComment(done0LabelInstr, "done0Label");
+         }
+      }
+   else
+      {
+      auto done0LabelInstr = generateLabelInstruction(cg, TR::InstOpCode::label, node, done0Label);
```

Comment:
Please add a comment like `/* Result: 0, 1, or 2 */'.

Reply:
Added the comment.